neard/src/migrations.rs

```diff
@@ -0,0 +1,115 @@
+use near_store::{create_store, DBCol, StoreUpdate};
+use std::path::Path;
+
+fn apply_block_at_height(
+    store_update: &mut StoreUpdate,
+    chain_store: &mut ChainStore,
+    runtime_adapter: &dyn RuntimeAdapter,
+    block_height: BlockHeight,
+    shard_id: ShardId,
+) -> Result<(), near_chain::Error> {
+    let block_hash = chain_store.get_block_hash_by_height(block_height)?;
+    let block = chain_store.get_block(&block_hash)?.clone();
+    if block.chunks()[shard_id as usize].height_included != block_height {
+        return Ok(());
+    }
+
+    let prev_block = chain_store.get_block(&block.header().prev_hash())?.clone();
+    let mut chain_store_update = ChainStoreUpdate::new(chain_store);
+    let receipt_proof_response = chain_store_update.get_incoming_receipts_for_shard(
+        shard_id,
+        block_hash,
+        prev_block.chunks()[shard_id as usize].height_included,
+    )?;
+    let receipts = collect_receipts_from_response(&receipt_proof_response);
+    let chunk = chain_store.get_chunk(&block.chunks()[shard_id as usize].chunk_hash())?;
+
+    let apply_result = runtime_adapter
+        .apply_transactions(
+            shard_id,
+            &chunk.header.inner.prev_state_root,
+            block_height,
+            block.header().raw_timestamp(),
+            block.header().prev_hash(),
+            block.hash(),
+            &receipts,
+            &chunk.transactions,
+            &chunk.header.inner.validator_proposals,
+            prev_block.header().gas_price(),
+            chunk.header.inner.gas_limit,
+            &block.header().challenges_result(),
+            *block.header().random_value(),
+        )
+        .unwrap();
+    let (_, outcome_paths) = ApplyTransactionResult::compute_outcomes_proof(&apply_result.outcomes);
+
+    for (outcome_with_id, proof) in apply_result.outcomes.into_iter().zip(outcome_paths.into_iter())
+    {
+        let id = outcome_with_id.id;
+        let mut existing_outcomes = chain_store.get_outcomes_by_id(&id)?;
+        existing_outcomes.push(ExecutionOutcomeWithIdAndProof {
+            proof,
+            block_hash,
+            outcome_with_id,
+        });
+        store_update.set_ser(DBCol::ColTransactionResult, id.as_ref(), &existing_outcomes)?;
+    }
+    Ok(())
+}
+
+pub fn migrate_12_to_13(path: &String, near_config: &NearConfig) {
+    let store = create_store(path);
+    if !near_config.client_config.archive {
+        // Non archival node. Perform a simply migration without necessarily fixing the inconsistencies
+        // since the old data will be garbage collected in five epochs
+        let mut store_update = store.store_update();
+        for (key, value) in store.iter_without_rc_logic(DBCol::ColTransactionResult) {
+            let outcome = ExecutionOutcomeWithIdAndProof::try_from_slice(&value).unwrap();
+            store_update.set_ser(DBCol::ColTransactionResult, &key, &vec![outcome]).unwrap();
+        }
+        store_update.commit().unwrap();
```

Comment:
Is it too expensive to fix all of those on non-archival nodes ("only" 5 epochs of data)? It would keep the migration simpler and don't introduce a corner case when the node gets migrated with this code path and then the user decides to restart the node as archival

Reply:
> Is it too expensive to fix all of those on non-archival nodes ("only" 5 epochs of data)? It would keep the migration simpler and don't introduce a corner case when the node gets migrated with this code path and then the user decides to restart the node as archival

It would make sense if we have properly separation between validator nodes and rpc nodes. As it stands today, validator nodes do not really care about this column and it doesn't make sense for them to do the migration.